### PR TITLE
Improves `Cookie::forever()` method for 32-bit servers #2968

### LINKED ISSUE #2968: Create lifetime cookie returns a error
I have made a Cookie consent solution by my own with Kirby Cookies. However, when I use this:
```php
<?php Cookie::forever('consent', 'all'); ?>
```
It works on localhost, but not on the hosting for the customer (Host Europe), I get this error:
```
Argument 1 passed to Kirby\Http\Cookie::lifetime() must be of the type int, float given, called in .../kirby/src/Http/Cookie.php on line 45
```
But this works.
```php
<?php Cookie::set('consent', 'all'); ?>
```
As reference, please check this thread at the forum. Is there a way to avoid this? Not sure if this is a bug or only a slow hosting problem. However, if I use the forever function, the site will not work.

https://forum.getkirby.com/t/create-lifetime-cookie-fails/20228/4

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

Since we do not have a test facility on a 32-bit server, I had to do the test without using the `forever()` method.

## Related issues

- Fixes #2968

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Fixed code style issues with CS fixer and `composer fix`
- [x] Added in-code documentation (if needed)

### CODE REVIEW

File: src/Http/Cookie.php [hunk 1]
```diff
@@ -63,11 +63,15 @@ public static function set(string $key, string $value, array $options = []): boo
     /**
      * Calculates the lifetime for a cookie
      *
-     * @param int $minutes Number of minutes or timestamp
+     * @param int|float $minutes Number of minutes or timestamp
      * @return int
      */
-    public static function lifetime(int $minutes): int
+    public static function lifetime($minutes): int
```
Comment (afbora) [inline src/Http/Cookie.php]: Removed `int` type because it converts to float if the server doesn't support it.

Comment (distantnative) [reply src/Http/Cookie.php]: What converts to float?

Comment (afbora) [reply src/Http/Cookie.php]: a number greater than the maximum integer, for 32-bit server on:

    Cookie::lifetime(253402214400); // you'll get type error

Comment (lukasbestle) [reply src/Http/Cookie.php]: To be honest I wouldn't allow floats, it would create a type mess that's very hard to test for edge-cases.

Have you tried the [solution I proposed here](https://github.com/getkirby/kirby/issues/2968#issuecomment-738266254) directly in the `forever()` method?

Comment (afbora) [reply src/Http/Cookie.php]: It works, but I wanted to combine it because the `lifetime()` method has the same issue. Of course, `forever()` Kirby originated, `lifetime()` is a user issue.

File: tests/Http/CookieTest.php [hunk 2]
```diff
@@ -40,6 +40,23 @@ public function testRemove()
...
+     * - 64-bit 253402214400
+     */
+    public function testForeverOn32BitServer()
+    {
+        // to be tested manually by sending the max integer on a 32-bit server
+        Cookie::set('forever', 'bar', ['lifetime' => 2147483647]);
+
+        $this->assertSame('703a07dc4edca348cb92d9fcb7da1b3931de0a85+bar', $_COOKIE['forever']);
+        $this->assertTrue(Cookie::exists('forever'));
+    }
```
Comment (afbora) [inline tests/Http/CookieTest.php]: You know, it's not actually a 32-bit test 🙈 Let's remove it if you wish.

Comment (lukasbestle) [reply tests/Http/CookieTest.php]: Hm, good point. 32-bit is not really testable without a 32-bit PHP binary. But I'd say it's fine as the year 2038 is coming closer and closer. 😜
Just kidding: It's an edge-case and we have now verified it manually and fixed it. So we don't really need a test if it's not viable (which it isn't).

File: src/Http/Cookie.php (hunk 1 above)
Comment (lukasbestle) [reply src/Http/Cookie.php]: That's a good point! I'm kind of split here – your original solution definitely fixes the issue well, but I'm not sure if we should weaken our type hints just for that edge-case. If we could already use union types (`int|float`) in PHP 7.3, it would be a non-issue.

**Review by lukasbestle — APPROVED** [review]
